modules/core/src/v2/coins/algo.ts

```diff
@@ -228,104 +236,113 @@ export class Algo extends BaseCoin {
     const factory = accountLib.getBuilder(this.getBaseChain()) as unknown as accountLib.Algo.TransactionBuilderFactory;
 
     const txBuilder = factory.from(txHex);
-    const tx = (await txBuilder.build()) as any;
+    const tx = (await txBuilder.build()) as Transaction;
     const txJson = tx.toJson();
 
-    if (tx.type === accountLib.BaseCoin.TransactionType.Send) {
-      const outputs: TransactionRecipient[] = [
-        {
-          address: txJson.to,
-          amount: txJson.amount,
-          memo: txJson.note,
-        },
-      ];
-      const operations: TransactionOperation[] = [];
-
-      const isTokenTx = this.isTokenTx(txJson.type);
-      if (isTokenTx) {
-        const type = accountLib.Algo.algoUtils.getTokenTxType(
-          txJson.amount,
-          txJson.from,
-          txJson.to,
-          txJson.closeRemainderTo
-        );
-        operations.push({
-          type: type,
-          coin: `${this.getChain()}:${txJson.tokenId}`,
-        });
-      }
-
-      const displayOrder = [
-        'id',
-        'outputAmount',
-        'changeAmount',
-        'outputs',
-        'changeOutputs',
-        'fee',
-        'memo',
-        'type',
-        'operations',
-      ];
-
-      const explanationResult: AlgoTransactionExplanation = {
-        displayOrder,
-        id: txJson.id,
-        outputAmount: txJson.amount.toString(),
-        changeAmount: '0',
-        outputs,
-        changeOutputs: [],
-        fee: txJson.fee,
-        memo: txJson.note,
-        type: tx.type,
-        operations,
-      };
-
-      if (txJson.tokenId) {
-        explanationResult.tokenId = txJson.tokenId;
-      }
+    switch (tx.type) {
+      case TransactionType.Send:
+      case TransactionType.EnableAsset:
+      case TransactionType.DisableAsset:
+        return this.explainSendTransaction(txJson);
+      case TransactionType.WalletInitialization:
+        return this.explainWalletInitializationException(txJson);
+    }
```

Comment:
you could avoid this switch with a simple if statement

```
if (tx.type === TransactionType.WalletInitialization) {
    return this.explainWalletInitializationException(txJson);
} else {
    return this.explainSendTransaction(txJson);
}
```

Reply:
They differ on the "default" case. The if creates a Send transaction while the switch is explicit on the handled cases. For a not handled type, it returns nothing (that was the original behavior).
But it should be better to be explicit on this behaviour

Reply:
improved